versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/util/SupplyOnce.java

```diff
@@ -41,27 +42,28 @@ private NonLockingSupplyOnce(Supplier<T> loader) {
     @Override
     public T get() {
       if (loaded) {
-        return eval(result);
+        return eval();
       }
 
       try {
         T obj = loader.get();
         result = obj;
         return obj;
       } catch (RuntimeException re) {
-        result = re;
+        ex = re;
         throw re;
       } finally {
         loaded = true;
       }
     }
 
     @SuppressWarnings("unchecked")
-    private T eval(Object i) {
-      if (i instanceof RuntimeException) {
-        throw (RuntimeException) i;
+    private T eval() {
+      Object r = result;
+      if (r != null) {
+        return (T) r;
       }
-      return (T) i;
+      throw ex;
```

Comment:
I'm not sure the order of assignments between `results`, `ex` and `loaded` is actually guaranteed to be the the same in another thread :thinking:

Reply:
This thing is not thread safe and not used in such a case

Reply:
There wasn't any synchronization in this class, so I am assuming that it is not meant to be thread-safe. Even the previous version would have had issues if `get()` is accessed concurrently.